### PR TITLE
Introduce IdentityStartup

### PR DESCRIPTION
**Pull request by hishamco**
Address https://github.com/oqtane/oqtane.framework/discussions/903

- Ability to toggle Identity service on / off

- Ability to replace the default Identity service implementation

### CODE REVIEW

File: Oqtane.Server/Startup.cs [hunk 1]
```diff
@@ -136,10 +140,10 @@ public void ConfigureServices(IServiceCollection services)
                 ));
             services.AddDbContext<TenantDBContext>(options => { });
 
-            services.AddIdentityCore<IdentityUser>(options => { })
-                .AddEntityFrameworkStores<TenantDBContext>()
-                .AddSignInManager()
-                .AddDefaultTokenProviders();
+            if (FeatureManager.IsEnabled("Identity"))
+            {
+                services.AddSingleton<IServerStartup, IdentityStartup>();
```
Comment (hishamco) [inline Oqtane.Server/Startup.cs:145]: Is this should be Singleton?

**Comment by hishamco** [comment]
@sbwalker this is a proof of concept of what you are suggested about using `IServerStartup` for the core services, I'm still vote to introduce a new interface `IIdentityStartup` and the default implementation will be `IdentityStartup` or provide a way to extend `IdentityStartup` by overriding the default method implementation

If you have another idea that ensure the registration order will not be changes, please shoot it to me

**Comment by hishamco** [comment]
What if we did something like this:

```csharp
services.AddSingleton<IdentityStartup>();
```

Then if we want a custom implementation

```csharp
public class PasswordLessIdentityStartup : IServerStartup
{
    public void Configure(IApplicationBuilder app, IWebHostEnvironment env)
    {

    }

    public void ConfigureMvc(IMvcBuilder mvcBuilder)
    {

    }

    public void ConfigureServices(IServiceCollection services)
    {
        services.AddSingleton<IdentityStartup, PasswordLessIdentityStartup>();
        services.AddIdentityCore<IdentityUser>()
            .AddEntityFrameworkStores<TenantDBContext>()
            .AddSignInManager()
            .AddDefaultTokenProviders()
            .AddPasswordlessLoginTotpTokenProvider();
    }
}
```

Is this will work?!!

File: Oqtane.Server/Startup.cs (hunk 1 above)
Comment (hishamco) [reply Oqtane.Server/Startup.cs:145]: Nah seems I did it wrong, the `IServerStartup` will register it's services automatically, but how we can controller the registration in `Startup`?

**Comment by sbwalker** [comment]
I understand the idea behind adding an interface for Identity, however there will be MANY different types of services which need to be registered in startup in the future, so having to introduce a new interface for each new scenario is simply not scalable. I would prefer to try and come up with a different approach.

**Comment by hishamco** [comment]
> I would prefer to try and come up with a different approach

That's why I asked in the discussion, how this should be implemented taking in mind the priority of activating the services

**Pull request closed without merging**